src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -1138,6 +1133,34 @@ public static bool IsParameterTypeContext(this SyntaxTree syntaxTree, int positi
             return false;
         }
 
+        public static bool IsPossibleExtensionMethodContext(this SyntaxTree syntaxTree, SyntaxToken tokenOnLeftOfPosition)
+        {
+            var method = tokenOnLeftOfPosition.GetAncestor<MethodDeclarationSyntax>();
+            var typeDecl = method.GetAncestorOrThis<TypeDeclarationSyntax>();
+
+            if (method == null || typeDecl == null)
+            {
+                return false;
+            }
+
+            if (typeDecl.Kind() != SyntaxKind.ClassDeclaration)
+            {
+                return false;
+            }
+
+            if (!method.Modifiers.Any(t => t.Kind() == SyntaxKind.StaticKeyword))
+            {
+                return false;
+            }
+
+            if (!typeDecl.Modifiers.Any(t => t.Kind() == SyntaxKind.StaticKeyword))
+            {
+                return false;
+            }
+
+            return true;
```

Comment:
could consdier simplifying this one big && expression.  it seems pretty clear to me.  i.e. `return method != null && typeDecl != null && typeDecl is a class, and type decl is static and method decl is static`.

Reply:
it's all just a narrowing set of simply truthy checks.  but your call.

Reply:
Will do (but note that this was just copied from ThisKeywordRecommender.cs

Reply:
➡️ Fixed in f2f612ec717d6920e5653c49acb42931dffcf81e